src/components/calcite-input-message/calcite-input-message.tsx

```diff
@@ -47,13 +47,21 @@ export class CalciteInputMessage {
   connectedCallback(): void {
     this.status = getElementProp(this.el, "status", this.status);
     this.scale = getElementProp(this.el, "scale", this.scale);
+    this.requestedIcon = setRequestedIcon(this.iconDefaults, this.icon, this.status);
```

Comment:
It seems like `requestedIcon` is only modified when the component is added to the DOM or moved within the DOM. What if the `icon` prop changes at after componentConnected? Should it recompute the requestedIcon?

Maybe the requestedIcon should also be set when either `icon` or `status` change?

Reply:
maybe this should be a `@State iconEl: HTMLCalciteIconElement;`?

```tsx
@Watch("status")
@Watch("icon")
handleIconEl(): void {
  this.iconEl = this.requestedIcon ? this.renderIcon(this.requestedIcon) : this.renderIcon(this.iconDefaults[this.status]);
}
```

Reply:
Nice, I'll try this approach - thank you for the insight @driskull!